src/chainrules.jl

```diff
@@ -0,0 +1,20 @@
+using ChainRulesCore
+
+function ChainRulesCore.ProjectTo(x::Union{KaNda, NdaKa})
+    return ProjectTo{KeyedArray}(;data=ProjectTo(keyless_unname(x)), keys=named_axiskeys(x))
+end
+
+function ChainRulesCore.ProjectTo(x::KeyedArray)
+    return ProjectTo{KeyedArray}(;data=ProjectTo(keyless(x)), keys=axiskeys(x))
+end
+
+(project::ProjectTo{KeyedArray})(dx) = wrapdims(project.data(dx), project.keys...)
```

Comment:
These need some direct tests of there own I think.

Testing that the keys of the result of `ProjectTo(ka)(dk)` for some plain `dk::Matrix`  and some `KeyedArray` `ka`, match the keys of `ka`

Reply:
added some unit tests